src/components/Common/UI/Switch/Switch.tsx

```diff
@@ -0,0 +1,74 @@
+import { Switch as _Switch } from 'react-aria-components'
+import classNames from 'classnames'
+import type { InputHTMLAttributes, FocusEvent, RefObject } from 'react'
+import { useFieldIds } from '../hooks/useFieldIds'
+import {
+  HorizontalFieldLayout,
+  type SharedHorizontalFieldLayoutProps,
+} from '../HorizontalFieldLayout'
+import styles from './Switch.module.scss'
+
+export interface SwitchProps
+  extends SharedHorizontalFieldLayoutProps,
+    Pick<InputHTMLAttributes<HTMLInputElement>, 'name' | 'id' | 'checked'> {
+  onBlur?: (e: FocusEvent) => void
+  onChange?: (checked: boolean) => void
+  inputRef?: RefObject<HTMLInputElement>
```

Comment:
In order for this to work with hook form, we'll need this to be able to handle a callback ref

that means this should be loosened to accept

```suggestion
  inputRef?: Ref<HTMLInputElement>
```

Unfortunately some of React Aria's components end up only accepting RefObject 😔 Hopefully that's not the case here. If it _is_ the case, i was doing the following

```
inputRef?: Ref<HTMLInputElement>

export function Switch({
  ...
  inputRef,
  ...
}: SwitchProps) {
  ...
  const internalInputRef = useRef<HTMLInputElement>(null)

  useEffect(() => {
    if (inputRef) {
      if (typeof inputRef === 'function') {
        inputRef(internalInputRef.current)
      } else {
        inputRef.current = internalInputRef.current
      }
    }
  }, [])
  return (
    <HorizontalFieldLayout
      ...
    />
      <_Switch
        ...
        inputRef={internalInputRef}
...
```

i ended up not having to use it because i ejected from the react aria implementations for checkbox and radio inputs but i had written this when i thought i needed to get around the fact that they only allowed a ref object

Reply:
Will be resolved in next iteration.